lib/commands/actions.js

```diff
@@ -1,23 +1,231 @@
+    } else {
+      log.warn(`Umount has failed, so not removing '${mntRoot}'`);
+    }
+  }
+}
+
+/**
+ * Get the content of given file from iOS Simulator and return it as base-64 encoded string.
+ *
+ * @param {Object} device - The device object, which represents the device under test.
+ *                          This object is expected to have the `udid` property containing the
+ *                          valid device ID.
+ * @param {string} remotePath - The path to a file, which exists in the corresponding application
+ *                              container on Simulator. The expected format of this string:
+ *                              @<app_bundle_id>/<path_to_the_file_inside_container>
+ * @returns {string} Base-64 encoded content of the file.
+ */
+async function pullFileFromSimulator (device, remotePath) {
+  if (!remotePath.startsWith(CONTAINER_PATH_MARKER)) {
+    log.errorAndThrow(`Only pulling files from application containers is supported for iOS Simulator. ` +
+                      `Provide the remote path in format @<bundle_id>/<path_to_the_file_in_its_container>`);
+  }
+  const bundleId = remotePath.substring(CONTAINER_PATH_MARKER.length, remotePath.indexOf('/'));
+  const containerRoot = await getAppContainer(device.udid, bundleId);
+  const dstPath = path.resolve(containerRoot, remotePath.substring(remotePath.indexOf('/') + 1));
+  log.info(`Parsed bundle identifier '${bundleId}' from '${remotePath}'. ` +
+           `Will get the data from '${dstPath}'`);
+  if (!await fs.exists(dstPath)) {
+    log.errorAndThrow(`The remote file at '${dstPath}' does not exist`);
+  }
+  const data = await fs.readFile(dstPath);
+  return new Buffer(data).toString('base64');
+}
+
+/**
+ * Get the content of given file from the real device under test and return it as base-64 encoded string.
+ *
+ * @param {Object} device - The device object, which represents the device under test.
+ *                          This object is expected to have the `udid` property containing the
+ *                          valid device ID.
+ * @param {string} remotePath - The path to an existing remote file on the device. This variable can be prefixed with
+ *                              bundle id, so then the file will be downloaded from the corresponding
+ *                              application container instead of the default media folder, for example
+ *                              '@com.myapp.bla/RelativePathInContainer/111.png'. The '@' character at the
+ *                              beginning of the argument is mandatory in such case.
+ * @return {string} Base-64 encoded content of the remote file
+ */
+async function pullFileFromRealDevice (device, remotePath) {
+  await verifyIFusePresence();
+  const mntRoot = await tempDir.tempDir();
+  let isUnmountSuccessful = true;
+  try {
+    let dstPath = path.resolve(mntRoot, remotePath);
+    let ifuseArgs = ['-u', device.udid, mntRoot];
+    if (remotePath.startsWith(CONTAINER_PATH_MARKER)) {
```

Comment:
This is beginning to be repetitive. The code for getting the bundleId and relative path could be extracted out.

Reply:
extracted